### PR TITLE
Ability to configure schedulerName for helm charts

### PR DESCRIPTION
**Pull request by StarZeus**
## Description
This adds an optional schedulerName in helm chart deployment allows users to configure it through values.yaml file.

Addresses: 
- Issue https://github.com/Budibase/budibase/issues/10017

## Documentation
- [x] I have reviewed the budibase documentatation to verify if this feature requires any changes. If changes or new docs are required I have written them.

### CODE REVIEW

File: charts/budibase/templates/minio-service-deployment.yaml [hunk 1]
```diff
@@ -72,6 +72,9 @@ spec:
       tolerations:
         {{- toYaml . | nindent 8 }}
     {{- end }}
+      {{- with .Values.schedulerName }}
```
Comment (Rory-Powell) [inline charts/budibase/templates/minio-service-deployment.yaml:75]: I think using `with` here might not be the correct syntax. From having a look at: https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with

I see that the `with` will rewrite `.` to mean the value of `.Values.schedulerName`. In this case that would mean the inner usage of `.Values.schedulerName` wouldn't exist. 

Two options
- We can switch the usages in this pr over to `if` instead
- Update the inner `.Values.schedulerName` in each case to just be `.`

**Review by Rory-Powell — CHANGES_REQUESTED** [review]
Comment on usage of `with`

File: charts/budibase/templates/minio-service-deployment.yaml (hunk 1 above)
Comment (StarZeus) [reply charts/budibase/templates/minio-service-deployment.yaml]: `with` is not required in these cases. So changed it to `if`. It should be good now.

**Review by Rory-Powell — APPROVED** [review]

**Comment by Rory-Powell** [comment]
@StarZeus I've merged this to develop to bring in your changes to our development cycle. If you wish, cherry-pick the commit and raise as a new pr to master - we can get it released a bit sooner then

**Pull request merged**